run/http_runner.go

```diff
@@ -0,0 +1,264 @@
+func SetAddr[Input, Option, Solution any](addr string) func(
+	*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) { r.setHTTPAddr(addr) }
+}
+
+// SetLogger sets the logger of the http server.
+func SetLogger[Input, Option, Solution any](l *log.Logger) func(
+	*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) { r.setLogger(l) }
+}
+
+// SetMaxParallel sets the maximum number of parallel requests.
+func SetMaxParallel[Input, Option, Solution any](maxParallel int) func(
+	*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) {
+		r.setMaxParallel(maxParallel)
+	}
+}
+
+// SetHTTPRequestHandler sets the function that handles the http request.
+func SetHTTPRequestHandler[Input, Option, Solution any](
+	f HTTPRequestHandler) func(*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) {
+		r.setHTTPRequestHandler(f)
+	}
+}
+
+// SetRunnerOption sets a runner option on the underlying runner.
+func SetRunnerOption[Input, Option, Solution any](
+	option RunnerOption[HTTPRunnerConfig, Input, Option, Solution],
+) func(
+	*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) {
+		r.setRunnerOption(option)
+	}
+}
+
+// SetHTTPServer sets the http server. Note that if you want to set the address
+// or the logger of the http server you are setting through this option and you
+// want to make use of SetAddr and SetLogger, you should pass them after passing
+// this option. Alternatively you can fully configure the http server and just
+// pass it to SetHTTPServer.
+func SetHTTPServer[Input, Option, Solution any](
+	s *http.Server) func(*httpRunner[Input, Option, Solution],
+) {
+	return func(r *httpRunner[Input, Option, Solution]) {
+		r.setHTTPServer(s)
+	}
+}
+
+// HTTPRunner is a runner that runs an algorithm as an http server.
+type HTTPRunner[RunnerConfig, Input, Option, Solution any] interface {
+	Runner[RunnerConfig, Input, Option, Solution]
+	// ActiveRuns returns the number of currently active runs.
+	ActiveRuns() int
+}
+
+// NewHTTPRunner creates a new NewHTTPRunner.
+func NewHTTPRunner[Input, Option, Solution any](
+	algorithm Algorithm[Input, Option, Solution],
+	options ...HTTPRunnerOption[Input, Option, Solution],
+) HTTPRunner[HTTPRunnerConfig, Input, Option, Solution] {
+	runner := &httpRunner[Input, Option, Solution]{
+		// the IOProducer will be dynamically set by the http request handler.
+		Runner: GenericRunner[HTTPRunnerConfig](
+			nil,
+			GenericDecoder[Input](decode.JSON()),
+			QueryParamDecoder[Option],
+			algorithm,
+			GenericEncoder[Solution, Option](encode.JSON()),
+		),
+	}
+
+	runnerConfig := runner.Runner.RunnerConfig()
+	runner.maxParallel = make(chan struct{}, runnerConfig.Runner.HTTP.MaxParallel)
+
+	// default http server
+	runner.httpServer = &http.Server{
```

Comment:
You need to set a `ReadHeaderTimeout` here, otherwise the server is vulnerable to the [Slowloris Attack](https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6).
Flagged by `gosec`: 
```
G112 (CWE-400): Potential Slowloris Attack because ReadHeaderTimeout is not configured in the http.Server (Confidence: LOW, Severity: MEDIUM)
```

Reply:
Thanks! I'll put in a fix.